application/models/Template.php

```diff
@@ -414,23 +411,31 @@ public static function getTemplatesFileFolder($sTemplateName = "")
     }
 
     /**
-     * Returns an array of all available template names - does a basic check if the template might be valid
-     *
-     * TODO: replace the calls to that function by a data provider based on search
-     *
-     * @return array
+     * Returns an array of all available template names - check if template exist
+     * key is template name, value is template folder
+     * @return string|]
      */
     public static function getTemplateList()
     {
-
-
-        $aTemplateList = array();
-
-        $oTemplateList = TemplateConfiguration::model()->search();
-        $oTemplateList->setPagination(false);
-
-        foreach ($oTemplateList->getData() as $oTemplate) {
-            $aTemplateList[$oTemplate->template_name] =  (self::isStandardTemplate($oTemplate->template_name)) ?  Yii::app()->getConfig("standardthemerootdir") . DIRECTORY_SEPARATOR . $oTemplate->template->folder : Yii::app()->getConfig("userthemerootdir") . DIRECTORY_SEPARATOR . $oTemplate->template->folder;
+        static $aTemplateList =  null;
```

Comment:
Usually, I don't like that much static variables inside methods. We already had a problem in the past when that variable needs to be refreshed. Can that static be outside of the method? Then we can have an extra method to refresh it if need to.

Reply:
Seems to be updated for master then : https://github.com/LimeSurvey/LimeSurvey/pull/3738

But it's a static variable inside a static function, i think it's OK.